pkg/resources/secret_with_oauth_client_credentials_acceptance_test.go

```diff
@@ -54,46 +58,117 @@ func TestAcc_SecretWithClientCredentials_BasicFlow(t *testing.T) {
+					assert.Check(resource.TestCheckTypeSetElemAttr(secretName, "oauth_scopes.*", "test")),
+				),
+			},
+			// set oauth_scopes and comment externally
+			{
+				PreConfig: func() {
+					req := sdk.NewAlterSecretRequest(id).WithSet(*sdk.NewSecretSetRequest().
+						WithSetForFlow(*sdk.NewSetForFlowRequest().WithSetForOAuthClientCredentials(
+							*sdk.NewSetForOAuthClientCredentialsRequest().WithOauthScopes(
+								*sdk.NewOauthScopesListRequest([]sdk.ApiIntegrationScope{{Scope: "bar"}}),
+							),
+						)),
+					)
+					acc.TestClient().Secret.Alter(t, req)
+				},
+				Config: config.FromModel(t, secretModelFooInScopesWithComment),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(secretModel.ResourceReference(), plancheck.ResourceActionUpdate),
+						planchecks.ExpectDrift(secretModel.ResourceReference(), "oauth_scopes", sdk.String("[test]"), sdk.String("[bar]")),
+						planchecks.ExpectChange(secretModel.ResourceReference(), "oauth_scopes", tfjson.ActionUpdate, sdk.String("[bar]"), sdk.String("[foo]")),
+					},
+				},
 				Check: assert.AssertThat(t,
 					resourceassert.SecretWithClientCredentialsResource(t, "snowflake_secret_with_client_credentials.s").
 						HasNameString(name).
 						HasDatabaseString(id.DatabaseName()).
 						HasSchemaString(id.SchemaName()).
 						HasApiAuthenticationString(integrationId.Name()).
+						HasOauthScopesLength(len([]string{"foo"})).
 						HasCommentString(newComment),
-					assert.Check(resource.TestCheckResourceAttr("snowflake_secret_with_client_credentials.s", "oauth_scopes.#", "1")),
-					assert.Check(resource.TestCheckTypeSetElemAttr("snowflake_secret_with_client_credentials.s", "oauth_scopes.*", "foo")),
+					assert.Check(resource.TestCheckResourceAttr(secretName, "oauth_scopes.#", "1")),
+					assert.Check(resource.TestCheckTypeSetElemAttr(secretName, "oauth_scopes.*", "foo")),
 				),
 			},
 			// unset comment
 			{
-				Config: config.FromModel(t, secretModelWithoutComment.WithOauthScopes([]string{"foo"})),
+				Config: config.FromModel(t, secretModelFooInScopes),
 				ConfigPlanChecks: resource.ConfigPlanChecks{
 					PreApply: []plancheck.PlanCheck{
-						planchecks.ExpectChange(secretModelWithoutComment.ResourceReference(), "comment", tfjson.ActionUpdate, sdk.String(newComment), nil),
+						plancheck.ExpectResourceAction(secretModelFooInScopes.ResourceReference(), plancheck.ResourceActionUpdate),
+						planchecks.ExpectChange(secretModelFooInScopes.ResourceReference(), "comment", tfjson.ActionUpdate, sdk.String(newComment), nil),
 					},
 				},
 				Check: assert.AssertThat(t,
-					resourceassert.SecretWithClientCredentialsResource(t, secretModelWithoutComment.ResourceReference()).
+					resourceassert.SecretWithClientCredentialsResource(t, secretModelFooInScopes.ResourceReference()).
 						HasCommentString(""),
 				),
 			},
-			// destroy
+			// unset comment externally
```

Comment:
SET comment externally. We don't have one in the config, and we change the comment in `PreConfig`.